### PR TITLE
feat: Implement support list of primitives saving in realm

### PR DESCRIPTION
**Pull request by yuliya-guseva**
#### Description
Implement  saving list of primitives to realm

#### Changes
Update creating realm scheme, saving object to realm, getting data from realm to object

#### Tests
Instrumented, Manual

### CODE REVIEW

File: android-lib/src/main/java/com/kinvey/android/cache/QueryHelper.java [hunk 1]
```diff
@@ -118,6 +122,19 @@ private static void in(RealmQuery query, String field, Object params){
         }
     }
 
+    private static void newIn(RealmQuery query,  String field, Object params){
+        query.beginGroup();
+        if (params.getClass().isArray()) {
+            String[] components = (String[]) params;
```
Comment (vinaygahlawat) [inline android-lib/src/main/java/com/kinvey/android/cache/QueryHelper.java]: If this is not part of the fix for saving and finding collections in Realm, then let's leave it out of this PR.

File: android-lib/src/androidTest/java/com/kinvey/androidTest/cache/CacheRealDataTest.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        phones.add("987654321");
+        person.setPhones(phones);
+        DefaultKinveyClientCallback saveCallback = testManager.save(store, person);
+        assertNotNull(saveCallback.getResult());
+        assertNotNull(saveCallback.getResult().getPhones());
+        assertTrue(saveCallback.getResult().getPhones().size() == 2);
+    }
+
+    @Test
+    public void testSaveArrayOfStringToRealmLocally() throws InterruptedException {
```
[the remark this answers is not in the document]
Comment (yuliya-guseva) [reply android-lib/src/androidTest/java/com/kinvey/androidTest/cache/CacheRealDataTest.java]: You are right. Renamed.

File: android-lib/src/main/java/com/kinvey/android/cache/QueryHelper.java [hunk 3]
```diff
@@ -17,6 +17,8 @@
 package com.kinvey.android.cache;
 
 
+import android.util.Log;
```
[the remark this answers is not in the document]
Comment (yuliya-guseva) [reply android-lib/src/main/java/com/kinvey/android/cache/QueryHelper.java]: Removed.

File: android-lib/src/main/java/com/kinvey/android/cache/QueryHelper.java (hunk 1 above)
Comment (yuliya-guseva) [reply android-lib/src/main/java/com/kinvey/android/cache/QueryHelper.java]: Removed.

**Pull request merged**